storagetransfer/storagetransfer_test.go

```diff
@@ -0,0 +1,266 @@
+	ctx := context.Background()
+
+	buf := new(bytes.Buffer)
+	resp, err := quickstart(buf, tc.ProjectID, gcsSourceBucket, gcsSinkBucket)
+
+	if err != nil {
+		t.Errorf("quickstart: %#v", err)
+	}
+
+	got := buf.String()
+	if want := "transferJobs/"; !strings.Contains(got, want) {
+		t.Errorf("quickstart: got %q, want %q", got, want)
+	}
+
+	tj := &storagetransferpb.TransferJob{
+		Name:   resp.Name,
+		Status: storagetransferpb.TransferJob_DELETED,
+	}
+	sts.UpdateTransferJob(ctx, &storagetransferpb.UpdateTransferJobRequest{
+		JobName:     resp.Name,
+		ProjectId:   tc.ProjectID,
+		TransferJob: tj,
+	})
+}
+
+func TestTransferFromAws(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	ctx := context.Background()
+
+	buf := new(bytes.Buffer)
+
+	resp, err := transfer_from_aws(buf, tc.ProjectID, "job description", s3Bucket, gcsSinkBucket, time.Now(), os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_KEY"))
+
+	if err != nil {
+		t.Errorf("transfer_from_aws: %#v", err)
+	}
+
+	got := buf.String()
+	if want := "transferJobs/"; !strings.Contains(got, want) {
+		t.Errorf("transfer_from_aws: got %q, want %q", got, want)
+	}
+
+	tj := &storagetransferpb.TransferJob{
+		Name:   resp.Name,
+		Status: storagetransferpb.TransferJob_DELETED,
+	}
+	sts.UpdateTransferJob(ctx, &storagetransferpb.UpdateTransferJobRequest{
+		JobName:     resp.Name,
+		ProjectId:   tc.ProjectID,
+		TransferJob: tj,
+	})
+}
+
+func TestTransferToNearline(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	ctx := context.Background()
+
+	buf := new(bytes.Buffer)
+
+	resp, err := transfer_to_nearline(buf, tc.ProjectID, "job description", gcsSourceBucket, gcsSinkBucket, time.Now())
+
+	if err != nil {
+		t.Errorf("transfer_from_aws: %#v", err)
+	}
+
+	got := buf.String()
+	if want := "transferJobs/"; !strings.Contains(got, want) {
+		t.Errorf("transfer_to_nearline: got %q, want %q", got, want)
+	}
+
+	tj := &storagetransferpb.TransferJob{
+		Name:   resp.Name,
+		Status: storagetransferpb.TransferJob_DELETED,
+	}
+	sts.UpdateTransferJob(ctx, &storagetransferpb.UpdateTransferJobRequest{
+		JobName:     resp.Name,
+		ProjectId:   tc.ProjectID,
+		TransferJob: tj,
+	})
+}
+
+func TestGetLatestTransferOperation(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	ctx := context.Background()
+
+	buf := new(bytes.Buffer)
+
+	job, err := transfer_to_nearline(buf, tc.ProjectID, "job description", gcsSourceBucket, gcsSinkBucket, time.Now())
+
+	op, err := check_latest_transfer_operation(buf, tc.ProjectID, job.Name)
+
+	if err != nil {
+		t.Errorf("check_latest_transfer_operation: %#v", err)
+	}
+	if !strings.Contains(op.Name, "transferOperations/") {
+		t.Errorf("check_latest_transfer_operation: Operation returned didn't have a valid operation name: %q", op.Name)
+	}
+
+	got := buf.String()
+	if want := op.Name; !strings.Contains(got, want) {
+		t.Errorf("check_latest_transfer_operation: got %q, want %q", got, want)
+	}
+	fmt.Println(got)
+
+	tj := &storagetransferpb.TransferJob{
```

Comment:
Could you factor out this cleanup logic into a separate function? It could even be called at the beginning of each test with `defer`